adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -1301,6 +1304,29 @@
             return promise.promise;
         }
 
+        function checkIfRealmAvailable() {
+            return new Promise(function (resolve, reject) {
```

Comment:
Please use `createPromise()` function similarly as in other functions.

Reply:
Ok, let it be, no problem. As stated in source code, KeycloakPromise is deprecated in favor to native Promise, so this abstraction looks useless for me and I used native.
BTW, I am surprised why there is no universal XHR function, and XHR is inlined in every place, so it's hard to introduce some common behavior for XHR, for example it is very necessary option to configure network connection timeouts for XHR.

Reply:
Yes you are right KeycloakPromise is deprecated, but deprecated doesn't mean removed yet. This way we can switch to native promises with a change just in one place in the future.

I agree with you. What you described sounds pretty good, I would be really happy to see this in the JS adapter in the future. If you would like to contribute this, feel free to create an enhancement Jira and provide a PR.